src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/DefaultTagHelperDescriptionFactory.cs

```diff
@@ -225,16 +228,20 @@ internal static bool TryExtractSummary(string documentation, out string summary)
                 return false;
             }
 
-            var summaryTagStart = documentation.IndexOf(summaryStartTag, StringComparison.OrdinalIgnoreCase);
-            if (summaryTagStart == -1)
-            {
-                summary = null;
-                return false;
-            }
+            documentation = documentation.Trim(NewLineChars);
 
+            var summaryTagStart = documentation.IndexOf(summaryStartTag, StringComparison.OrdinalIgnoreCase);
             var summaryTagEndStart = documentation.IndexOf(summaryEndTag, StringComparison.OrdinalIgnoreCase);
-            if (summaryTagEndStart == -1)
+            if (summaryTagStart == -1 || summaryTagEndStart == -1)
             {
+                // A really wrong but cheap way to check if this is XML
+                if ((!documentation.StartsWith("<") && !documentation.EndsWith(">")) && (summaryTagStart == -1 && summaryTagEndStart == -1))
```

Comment:
What does the second part of this if statement buy us? Wondering if we're being too protective is all :)

Reply:
The idea here is that when someone has a malformed `<summary>`, IE:
```
<summary>But no ending tag
```

we previously counted this as "return nothing" and I wanted to minimize the changes to existing scenarios. Without the `(summaryTagStart == -1 && summaryTagEndStart == -1)` we would start to return the whole value in that case, which felt wrong.

Reply:
Ah so malformed `<summary>` tags. Would it break anything if we returned them with `<summary>But no ending tag`? Wondering if we're protecting TagHelper authors unnecessarily where the real fix would be for the TagHelper author to provide a valid XML doc

Reply:
I don't think it would break anything, I was just protecting the existing behavior. If we'd prefer to surface malformed XML I can totally remove that conditional.

Reply:
I'd allow the malformed XML to be honest. That way users know if they have mistakes. Otherwise we hide it 😄